src/uwtools/rocoto.py

```diff
@@ -97,24 +105,25 @@ def dump(self, path: OptionalPath = None) -> None:
         with writable(path) as f:
             f.write(str(self).strip())
 
-    def _add_compound_time_string(self, e: Element, config: dict, tag: str) -> None:
+    def _add_compound_time_string(self, e: Element, config: Any, tag: str) -> Element:
```

Comment:
In order to support `<datadep>` and `<timedep>` correctly, callers need to pass in the required subconfig now. And the value could be (for e.g. `<timedep>`) an `int` as well as a `str`, so I've punted and typed `config` as `Any`. Note below that the value is cast to a `str` before being set to the XML elements text field.

Also, the created child element is now returned so that the caller can modify it further -- e.g. the handler for `<datadep>` elements adds attributes to it.